osquery/core/windows/handle.cpp

```diff
@@ -0,0 +1,118 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <Windows.h>
+
+#include "osquery/core/windows/ntapi.h"
+#include "osquery/core/windows/handle.h"
+
+namespace osquery {
+namespace tables {
+
+Handle::Handle() {
+  _h = NULL;
+}
+
+Handle::~Handle() {
+  close();
+}
+
+HANDLE Handle::getAsHandle() {
+  return _h;
+}
+
+bool Handle::valid() {
+  return (NULL != _h);
+}
+
+void Handle::close() {
+  if (valid()) {
+    CloseHandle(_h);
+    _h = NULL;
+  }
+}
+
+bool Handle::openSymLinkObj(const std::wstring& strName) {
```

Comment:
Looks like this should be `Status Handle::openSymLinkObj(const std::wstring& strName)` as you that's how you have it declared and seems to be how you're expecting to use it in your table code. Perhaps this could `return Status(1, "Open symbolic link to object failed with " + std::to_string(GetLastError()));` or `return Status();` otherwise?

Reply:
@muffins ok got it

Reply:
done